services/package-json-helpers.js

```diff
@@ -0,0 +1,54 @@
+'use strict'
+
+const Joi = require('joi')
+const { InvalidParameter } = require('./errors')
+
+const isDependencyMap = Joi.object()
+  .pattern(
+    /./,
+    // This accepts a semver range, a URL, and many other possible values.
+    Joi.string()
+      .min(1)
+      .required()
+  )
+  .default({})
+
+const isPackageJsonWithDependencies = Joi.object({
+  dependencies: isDependencyMap,
+  devDependencies: isDependencyMap,
+  peerDependencies: isDependencyMap,
+}).required()
+
+function getDependencyVersion({
+  kind = 'prod',
+  wantedDependency,
+  dependencies,
+  devDependencies,
+  peerDependencies,
+}) {
+  let dependenciesOfKind
+  if (kind === 'peer') {
+    dependenciesOfKind = peerDependencies
+  } else if (kind === 'dev') {
+    dependenciesOfKind = devDependencies
+  } else if (kind === 'prod') {
+    dependenciesOfKind = dependencies
+  } else {
+    throw Error(`Not very kind: ${kind}`)
```

Comment:
😆 The only time this would get hit is if a service accepts and/or passes along a bad dependency kind here right?

Reply:
Thought I responded to this before. Yes! That would be a programmer error.

Reply:
Okay cool. Just wanted to make sure we weren't planning on surfacing this message up to end users 
👍

Reply:
If it got thrown in production, they'd see **shields internal error**.